### PR TITLE
Use ipm distortion instead of image rectification

### LINKED ISSUE #363: Head mover takes very long time to react to head mode change
# Summary
While executing a head pattern the head mover is hard to stop `mode=7`. It takes many messages and a long time.

### PR DESCRIPTION
**Pull request by Flova**
# Summary
Consider distortion in the IPM instead of rectifying the whole image, as the image rectification degrades the image quality and takes a significant amount of time. Also switch from the Ros Pylon Driver to a custom node that communicates with pylon. The reasoning behind this is a hard to fix timestamp issue in the driver. Also gets rid of the blaze dependency and a number of compiler warnings the other maintainer refused to fix.

## Proposed changes
- Use new distortion feature in the IPM
- Remove rectification from image pipeline
- Add extrinsic calibration of imu
- Do image postprocessing in custom node instead of image_proc
- Add calibration for Amy and Rory
- Remove header from extrinsic calibration node (we will not link against it and it is more convenient that way
- Fix high load of head mover
- Use own pylon ros driver

## Related issues
Waits for https://github.com/ros-sports/ipm/pull/64 and the linked PRs there.
Closes #363

## Checklist

- [ ] Run `colcon build`
- [ ] Write documentation
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Create issues for future work
- [ ] Triage this PR and label it

### CODE REVIEW

File: bitbots_motion/bitbots_head_mover/src/move_head.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ class HeadMover {
 
     // Initialize subscriber for the current joint states of the robot
     joint_state_subscriber_ = node_->create_subscription<sensor_msgs::msg::JointState>(
-        "joint_states", 10, [this](const sensor_msgs::msg::JointState::SharedPtr msg) { joint_state_callback(msg); });
+        "joint_states", 1, [this](const sensor_msgs::msg::JointState::SharedPtr msg) { joint_state_callback(msg); });
```
Comment (jaagut) [inline bitbots_motion/bitbots_head_mover/src/move_head.cpp:105]: please PR description

File: bitbots_wolfgang/wolfgang_description/urdf/robot.urdf [hunk 2]
```diff
@@ -374,18 +374,19 @@
       </material>
     </visual>
   </link>
-  <link name="imu_frame">
+  <!-- Renamed due to extrinsic calibration -->
```
Comment (jaagut) [inline bitbots_wolfgang/wolfgang_description/urdf/robot.urdf:377]: please PR description

File: bitbots_misc/bitbots_extrinsic_calibration/src/extrinsic_calibration.cpp [hunk 3]
```diff
@@ -1,74 +1,82 @@
...
+class ExtrinsicCalibrationBroadcaster : public rclcpp::Node {
+ public:
+  ExtrinsicCalibrationBroadcaster() : Node("bitbots_extrinsic_calibration") {
+    broadcaster_ = std::make_unique<tf2_ros::StaticTransformBroadcaster>(this);
+
+    this->declare_parameter<std::string>("parent_frame", "camera_optical_frame_uncalibrated");
+    this->declare_parameter<std::string>("child_frame", "camera_optical_frame");
+    this->declare_parameter<double>("offset_x", 0.0, rcl_interfaces::msg::ParameterDescriptor());
+    this->declare_parameter<double>("offset_y", 0.0);
+    this->declare_parameter<double>("offset_z", 0.0);
```
Comment (jaagut) [inline bitbots_misc/bitbots_extrinsic_calibration/src/extrinsic_calibration.cpp:22]: What is parameter descriptor and why only in one line?

**Review by jaagut — COMMENTED** [review]
Did not properly read calibrator. Maybe someone else could have a look at it?

File: bitbots_wolfgang/wolfgang_description/urdf/robot.urdf (hunk 2 above)
Comment (Flova) [reply bitbots_wolfgang/wolfgang_description/urdf/robot.urdf:377]: Already there

File: bitbots_misc/bitbots_extrinsic_calibration/src/extrinsic_calibration.cpp (hunk 3 above)
Comment (Flova) [reply bitbots_misc/bitbots_extrinsic_calibration/src/extrinsic_calibration.cpp]: I don't know. It was there previously. I removed it, as it should not make any difference.

File: bitbots_misc/bitbots_ipm/launch/ipm.launch [hunk 4]
```diff
@@ -1,17 +1,18 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <launch>
     <arg name="sim" default="false"/>
-    <arg name="full_image" default="false"/>
-    <arg name="markers" default="false"/>
-    <arg name="rviz" default="false"/>
+    <arg name="full_image" default="false" description="Whether to project the full rgb image for debugging or shwcasing"/>
+    <arg name="markers" default="false" description="Whether to publish markers for visualization of the detected objects in rviz"/>
+    <arg name="rviz" default="false" description="Whether to start rviz with the ipm configuration"/>
```
Comment (jaagut) [inline bitbots_misc/bitbots_ipm/launch/ipm.launch:6]:
```suggestion
    <arg name="full_image" default="false" description="Whether to project the full-size RGB image for debugging or showcasing"/>
    <arg name="markers" default="false" description="Whether to publish markers for visualization of the detected objects in RViz"/>
    <arg name="rviz" default="false" description="Whether to start RViz with the ipm configuration"/>
```

File: bitbots_wolfgang/wolfgang_description/urdf/robot.urdf (hunk 2 above)
Comment (jaagut) [reply bitbots_wolfgang/wolfgang_description/urdf/robot.urdf:377]: My goal was to mention the new IMU (static) frames. We should document somewhere what the extrisic calibration does and where those frames magically come from...

**Review by jaagut — APPROVED** [review]

**Pull request merged**